### ISSUE TITLE
How to handle unreleased test dependencies?

### ISSUE THREAD

**Issue by JeroenDM** [JeroenDM/moveit_opw_kinematics_plugin#63]
The tests currently have a run dependency on [kuka_test_resources]() that is not released (and probably should not be released). This means the test will fail on the build farm.

Should we solve this problem? And if so, these are some ideas:
- Refactor the tests and load the urdf files directly from the test directory. Replace all meshes with primitive shapes.
- Use robots that have released moveit_config packages.
- Add kuka_test_resources to the external directory (not my favorite solution, but the easiest).

**Comment by JeroenDM** [comment]
For future reference:
- ROS answers [question](https://answers.ros.org/question/353824/using-unreleased-test-dependencies-on-the-ros-build-farm/?comment=353936#post-id-353936).
- moveit_resources [issue](https://github.com/ros-planning/moveit_resources/issues/27)

I'm really not sure anymore what would be a good solution.

The option mentioned above that would avoid much of the issues raised is to create a _minimal_ version of the urdf of the fanuc and kuka robot that only uses primitive shapes for links. These can then be but in the test folder and loaded to run tests. (Only the kinematic structure is relevant for the tests, so there should not even be a collision geometry in the urdf.)

This is of course quite some manual work to recreate resources that already exist anyway.

**Comment by gavanderhoorn** [comment]
So what would be needed to run the tests?

Could you "simply" `test_depend` on `fanuc_m10ia_support` fi?

**Comment by JeroenDM** [comment]
At the moment we use the moveit config package [here](https://github.com/JeroenDM/moveit_opw_kinematics_plugin/blob/abc361eb3fb719c4add60848e4edee1a508d0dca/test/test_fanuc.rostest#L4).

And looking at it again, I think this should stay this way.
Only loading the urdf for a test might not be a good test. By using the moveit config package we also test whether the plugin is loaded correctly, exactly replicating how it will mostly be used by users.

I think this package should test whether the plugin is loaded correctly and works from a variety of robots. What we should **not** specifically test:

- Checking the kinematics calculations (responsibility of the opw_kinematics library).
- Checking the geometric parameters for a specific robot (responsibility of the robot support package).

In conclusion, I think it can be useful to release the moveit_configurations for some robots, as is done now with `panda_moveit_config`. This is not only good for the tests here, but also for tests and tutorials in general.

Searching through the [ROS index](https://index.ros.org/search/?term=*moveit_config&section=pkgs), there are apparently already a whole bunch of these released (mostly fanuc and some abb). So, maybe I can just change the tests to use some of these robots... (Ideally while adding the opw geometric parameters to their support package.)

**Comment by gavanderhoorn** [comment]
> In conclusion, I think it can be useful to release the moveit_configurations for some robots, as is done now with `panda_moveit_config`. This is not only good for the tests here, but also for tests and tutorials in general.

that was going to be my next question: there are plenty of MoveIt configurations released.

You could use those.

You could also opt to release a specific test moveit configuration package you put together yourself. It would not need any meshes, nor other fancy stuff, so really only be minimal.

Given the requirement of having `test_depend`s to be released -- or at least when working with the ROS buildfarm -- you could then release that package and be done with this.

How much value there would be in testing whether the plugin loads I'm not sure though: there are various scenarios in which such a test would succeed in your test, but could still fail on users' machines.
